extensions/common/context.h

```diff
@@ -248,6 +258,12 @@ class PeerNodeInfo {
   flatbuffers::DetachedBuffer fallback_peer_node_;
 };
 
+// Populate shared information between all protocols.
+// Requires that the connections are established both downstrean and upstream.
+// Caches computation using is_populated field.
+void populateRequestInfo(bool outbound, bool use_host_header_fallback,
+                         RequestInfo* request_info);
+
 // populateHTTPRequestInfo populates the RequestInfo struct. It needs access to
 // the request context.
 void populateHTTPRequestInfo(bool outbound, bool use_host_header,
```

Comment:
nit: should this really be populateHTTPAndGrpcRequestInfo?

Reply:
We treat gRPC as a subset of HTTP throughout. I think it's OK for now, since it's guarded by protocol field.